language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/performance/CodeActionPerformanceTest.java

```diff
@@ -0,0 +1,194 @@
+import org.ballerinalang.langserver.util.TestUtil;
+import org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager;
+import org.eclipse.lsp4j.CodeActionContext;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.jsonrpc.Endpoint;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Test performance of code actions feature in language server.
+ */
+public class CodeActionPerformanceTest {
+    private Endpoint serviceEndpoint;
+    private final JsonParser parser = new JsonParser();
+    private final Path testRoot = FileUtils.RES_DIR.resolve("performance");
+    private static final WorkspaceManager workspaceManager
+            = BallerinaWorkspaceManager.getInstance(new LanguageServerContextImpl());
+    private static final LanguageServerContext serverContext = new LanguageServerContextImpl();
+
+    @BeforeClass
+    public void init() throws Exception {
+        this.serviceEndpoint = TestUtil.initializeLanguageSever();
+    }
+
+    @Test(dataProvider = "performance-data-provider")
+    public void testCodeAction(String config, String source) throws IOException, WorkspaceDocumentException {
+        String configJsonPath = getConfigJsonPath(config);
+        Path sourcePath = testRoot.resolve(getResourceDir()).resolve("source").resolve(source);
+        JsonObject configJsonObject = FileUtils.fileContentAsObject(configJsonPath);
+        TestUtil.openDocument(serviceEndpoint, sourcePath);
+
+        // Filter diagnostics for the cursor position
+        List<io.ballerina.tools.diagnostics.Diagnostic> diagnostics
+                = TestUtil.compileAndGetDiagnostics(sourcePath, workspaceManager, serverContext);
+        List<Diagnostic> diags = new ArrayList<>(CodeActionUtil.toDiagnostics(diagnostics));
+        Position pos = new Position(configJsonObject.get("line").getAsInt(),
+                configJsonObject.get("character").getAsInt());
+        diags = diags.stream()
+                .filter(diag -> CommonUtil.isWithinRange(pos, diag.getRange()))
+                .collect(Collectors.toList());
+        CodeActionContext codeActionContext = new CodeActionContext(diags);
+
+        Range range = new Range(pos, pos);
+        long start = System.currentTimeMillis();
+        String res = TestUtil.getCodeActionResponse(serviceEndpoint, sourcePath.toString(), range, codeActionContext);
+        long end = System.currentTimeMillis();
+        long responseTime = end - start;
+        Assert.assertEquals(responseTime < 3000, true);
```

Comment:
When checking a condition, we can use `Assert.assertTrue(responseTime < 3000)` instead. Also, move `3000` to a class level constant (static variable)